toucan_connectors/http_api/http_api_connector.py

```diff
@@ -19,26 +26,24 @@
 
 from toucan_connectors.auth import Auth
 from toucan_connectors.common import (
-    FilterSchema,
-    XpathSchema,
     nosql_apply_parameters_to_query,
     transform_with_jq,
 )
 from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
 from toucan_connectors.utils.json_to_table import json_to_table
 
+TOO_MANY_REQUESTS = 429
+
+
+class HttpAPIConnectorError(Exception):
+    """Raised when an error occurs while fetching data from an HTTP API"""
+
```

Comment:
would be great if we had an `original_exc` attribute for later investigation:

```suggestion
class HttpAPIConnectorError(Exception):
    """Raised when an error occurs while fetching data from an HTTP API"""
    
    def __init__(self, original_exc: HttpError) -> None:
        super().__init__()
        self.original_exc = original_exc

```